### PR TITLE
Add support for custom type inference through the plugin options

### LINKED ISSUE #29: Customizable inferJSONSchemaType
Make it able to add custom scalar inferring types, so far it only support 3 types

GraphQLString <=> { type: 'string' }
GraphQLInt <=> { type: 'integer' }
GraphQLFloat <=> { type: 'number' }

### PR DESCRIPTION
**Pull request by bilalshareef**
Fix #29

### CODE REVIEW

File: test/json-schema-validation.js [hunk 1]
```diff
@@ -2089,4 +2090,131 @@ t.test('JSON Schema validators', t => {
       }
     })
   })
+
+  t.test('should invoke inferCustomType option and not affect operations when everything is okay', async (t) => {
+    t.plan(1)
```
Comment (simoneb) [inline test/json-schema-validation.js:2095]: this is unnecessay

Comment (bilalshareef) [reply test/json-schema-validation.js]: I removed it now.

**Review by simoneb — CHANGES_REQUESTED** [review]
I tihnk we need to review the logic here. If somebody provides a custom inference function, we surely need to invoke it, but that doesn't necessarily mean that it will hande the scalar type we're currently processing. Meaning that we need to define AND document a signature for that function, because users need to know how to use it and what to return from it.

The behavior should be this:

- the custom function can override the default logic, meaning that it should be invoked first
- the signature should be so that if the custom function doesn't handle the type, it returns a falsy value, in which case the default logic kicks in

**Review by simoneb — APPROVED** [review]

**Review by jonnydgreen — APPROVED** [review]
LGTM - fantastic work!

**Comment by jonnydgreen** [comment]
Will aim to include as part of the next release if that works :)

**Pull request merged**